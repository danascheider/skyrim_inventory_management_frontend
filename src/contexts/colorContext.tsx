import { createContext, ReactElement } from 'react'
import { YELLOW, type ColorScheme } from '../utils/colorSchemes'

const ColorContext = createContext<ColorScheme>(YELLOW)

interface ProviderProps {
  colorScheme: ColorScheme
  children: ReactElement
}

const ColorProvider = ({ colorScheme, children }: ProviderProps) => (
  <ColorContext.Provider value={colorScheme}>{children}</ColorContext.Provider>
)

export { ColorProvider, ColorContext }
