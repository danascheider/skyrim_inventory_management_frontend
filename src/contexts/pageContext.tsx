import { createContext, useState, useEffect, useRef } from 'react'
import { type ApiCalls, type Resource, type HttpVerb } from '../types/apiCalls'
import { ProviderProps } from '../types/contexts'
import { type FlashProps, type ModalProps } from '../types/pageContext'

interface PageContextType {
  flashProps: FlashProps
  setFlashProps: (props: FlashProps) => void
  modalProps: ModalProps
  setModalProps: (props: ModalProps) => void
  apiCallsInProgress: ApiCalls
  addApiCall: (key: Resource, value: HttpVerb) => void
  removeApiCall: (key: Resource, value: HttpVerb) => void
}

const defaultFlashProps: FlashProps = {
  type: 'info',
  message: '',
  hidden: true,
}

const defaultModalProps: ModalProps = {
  hidden: true,
  children: <></>,
}

const defaultApiCallStatus = {
  games: [],
  shoppingLists: [],
  shoppingListItems: [],
}

const PageContext = createContext<PageContextType>({
  flashProps: defaultFlashProps,
  modalProps: defaultModalProps,
  apiCallsInProgress: defaultApiCallStatus,
  setFlashProps: () => {},
  setModalProps: () => {},
  addApiCall: () => {},
  removeApiCall: () => {},
})

const PageProvider = ({ children }: ProviderProps) => {
  const [flashProps, setFlashProps] = useState<FlashProps>(defaultFlashProps)
  const [modalProps, setModalProps] = useState<ModalProps>(defaultModalProps)
  const [apiCallsInProgress, SetApiCallsInProgress] =
    useState<ApiCalls>(defaultApiCallStatus)

  const flashVisibleSince = useRef(0)

  const addApiCall = (key: Resource, value: HttpVerb) => {
    SetApiCallsInProgress({
      ...apiCallsInProgress,
      [key]: [...apiCallsInProgress[key], value],
    })
  }

  const removeApiCall = (key: Resource, value: HttpVerb) => {
    SetApiCallsInProgress({
      ...apiCallsInProgress,
      [key]: apiCallsInProgress[key].filter((verb) => verb !== value),
    })
  }

  const value = {
    flashProps,
    modalProps,
    apiCallsInProgress,
    setFlashProps,
    setModalProps,
    addApiCall,
    removeApiCall,
  }

  useEffect(() => {
    if (flashProps.hidden === true) return

    flashVisibleSince.current = Number(new Date())

    setTimeout(() => {
      if (Number(new Date()) - flashVisibleSince.current >= 4000) {
        setFlashProps({ ...flashProps, hidden: true })
        flashVisibleSince.current = 0
      }
    }, 4000)
  }, [flashProps])

  return <PageContext.Provider value={value}>{children}</PageContext.Provider>
}

export { PageContext, PageProvider }
