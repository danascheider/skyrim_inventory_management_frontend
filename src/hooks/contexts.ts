import { useContext } from 'react'
import { ColorContext } from '../contexts/colorContext'
import { LoginContext } from '../contexts/loginContext'

const useCustomContext = <T>(cxt: React.Context<T>, msg: string) => {
  const context = useContext(cxt)

  if (!context) throw new Error(msg)

  return context
}

export const useColorScheme = () =>
  useCustomContext(
    ColorContext,
    'useColorScheme must be used within a ColorProvider'
  )

export const useGoogleLogin = () =>
  useCustomContext(
    LoginContext,
    'useGoogleLogin must be used within a LoginProvider'
  )