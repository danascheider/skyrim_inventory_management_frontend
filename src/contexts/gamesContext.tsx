import { createContext, useCallback, useEffect, useState } from 'react'
import { signOutWithGoogle } from '../firebase'
import { type Game } from '../types/games'
import { type ProviderProps } from '../types/contexts'
import { useGoogleLogin, usePageContext } from '../hooks/contexts'
import { ApiError } from '../types/errors'
import { LOADING, DONE, ERROR, type LoadingState } from '../utils/loadingStates'
import { getGames, deleteGame } from '../utils/simApi'

// TODO: Add default values for context provider as follows:
//       const createGame = (game: Game) => {
//                                            postGames(game: Game, token: '')
//                                              .then((status: number, json: string) => /* noop */)
//                                          }
//       This will be useful in Storybook.

const NOT_FOUND_MESSAGE =
  "Oops! We couldn't find the game you're looking for. Please refresh and try again."
const UNEXPECTED_ERROR_MESSAGE =
  "Oops! Something unexpected went wrong. We're sorry! Please try again later."

export interface GamesContextType {
  games: Game[]
  gamesLoadingState: LoadingState
  // createGame: (game: Game) => void
  // updateGame: (gameId: number, attrs: Game) => void
  destroyGame: (gameId: number) => void
}

export const GamesContext = createContext<GamesContextType>({
  games: [],
  gamesLoadingState: LOADING,
  destroyGame: () => {},
})

export const GamesProvider = ({ children }: ProviderProps) => {
  const { user, token, authLoading, requireLogin } = useGoogleLogin()
  const [gamesLoadingState, setGamesLoadingState] = useState(LOADING)
  const [games, setGames] = useState<Game[]>([])
  const { setFlashProps } = usePageContext()

  /*
   *
   * Retrieve all the current user's games from the API
   *
   */

  const fetchGames = useCallback(() => {
    if (user && token) {
      getGames(token)
        .then(({ json }) => {
          if (Array.isArray(json)) {
            setGames(json)
            setGamesLoadingState(DONE)
          }
        })
        .catch((e: ApiError) => {
          if (import.meta.env.DEV)
            console.error(`Error ${e.name}: ${e.message}`)

          if (e.code === 401) {
            signOutWithGoogle()
          } else {
            setGames([])
            setGamesLoadingState(ERROR)
            setFlashProps({
              type: 'error',
              message: e.message,
              hidden: false,
            })
          }
        })
    }
  }, [user, token])

  /*
   *
   * Destroy the requested game and update the `games` array
   *
   */

  const destroyGame = useCallback(
    (gameId: number) => {
      if (user && token) {
        deleteGame(gameId, token)
          .then(({ status }) => {
            if (status === 204) {
              const newGames = games.filter(({ id }) => id !== gameId)
              setGames(newGames)
              setFlashProps({
                hidden: false,
                type: 'success',
                message: 'Success! Your game has been deleted.',
              })
            }
          })
          .catch((e: ApiError) => {
            if (import.meta.env.DEV)
              console.log(`Error ${e.name}: ${e.message}`)

            const message =
              e.code === 404 ? NOT_FOUND_MESSAGE : UNEXPECTED_ERROR_MESSAGE

            setFlashProps({
              hidden: false,
              type: 'error',
              message,
            })
          })
      }
    },
    [user, token, games]
  )

  const value = {
    games,
    gamesLoadingState,
    destroyGame,
  }

  useEffect(() => {
    requireLogin()
  }, [requireLogin])

  useEffect(() => {
    if (authLoading) return

    fetchGames()
  }, [authLoading, fetchGames])

  return <GamesContext.Provider value={value}>{children}</GamesContext.Provider>
}
