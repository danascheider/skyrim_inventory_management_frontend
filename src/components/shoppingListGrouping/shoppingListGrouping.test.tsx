import { describe, test, expect } from 'vitest'
import { renderAuthenticated } from '../../support/testUtils'
import {
  gamesContextValue,
  shoppingListsContextValueEmpty,
} from '../../support/data/contextValues'
import { shoppingListsContextValue } from '../../support/data/contextValues'
import { PageProvider } from '../../contexts/pageContext'
import { GamesContext } from '../../contexts/gamesContext'
import { ShoppingListsContext } from '../../contexts/shoppingListsContext'
import ShoppingListGrouping from './shoppingListGrouping'

describe('ShoppingListGrouping', () => {
  describe('when there are shopping lists', () => {
    test('displays each list', () => {
      const wrapper = renderAuthenticated(
        <PageProvider>
          <GamesContext.Provider value={gamesContextValue}>
            <ShoppingListsContext.Provider value={shoppingListsContextValue}>
              <ShoppingListGrouping />
            </ShoppingListsContext.Provider>
          </GamesContext.Provider>
        </PageProvider>
      )

      expect(wrapper).toBeTruthy()

      expect(wrapper.getByText('All Items')).toBeTruthy()
      expect(wrapper.getByText('Honeyside')).toBeTruthy()
      expect(wrapper.getByText('Breezehome')).toBeTruthy()

      // There should be list items on each list
      expect(wrapper.getAllByText('Dwarven Cog').length).toEqual(3)
      expect(
        wrapper.getAllByText(
          'This item has a really really really really really long description for testing purposes'
        ).length
      ).toEqual(2)
    })

    test('matches snapshot', () => {
      const wrapper = renderAuthenticated(
        <PageProvider>
          <GamesContext.Provider value={gamesContextValue}>
            <ShoppingListsContext.Provider value={shoppingListsContextValue}>
              <ShoppingListGrouping />
            </ShoppingListsContext.Provider>
          </GamesContext.Provider>
        </PageProvider>
      )

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when there are no shopping lists', () => {
    test('displays a message that there are no lists for this game', () => {
      const wrapper = renderAuthenticated(
        <PageProvider>
          <GamesContext.Provider value={gamesContextValue}>
            <ShoppingListsContext.Provider
              value={shoppingListsContextValueEmpty}
            >
              <ShoppingListGrouping />
            </ShoppingListsContext.Provider>
          </GamesContext.Provider>
        </PageProvider>
      )

      expect(wrapper.getByText('This game has no shopping lists.')).toBeTruthy()
    })

    test('matches snapshot', () => {
      const wrapper = renderAuthenticated(
        <PageProvider>
          <GamesContext.Provider value={gamesContextValue}>
            <ShoppingListsContext.Provider
              value={shoppingListsContextValueEmpty}
            >
              <ShoppingListGrouping />
            </ShoppingListsContext.Provider>
          </GamesContext.Provider>
        </PageProvider>
      )

      expect(wrapper).toMatchSnapshot()
    })
  })
})
