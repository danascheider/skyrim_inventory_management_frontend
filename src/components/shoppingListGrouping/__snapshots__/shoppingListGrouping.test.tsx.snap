// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShoppingListGrouping > when there are no shopping lists > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="_noLists_92c834"
      >
        This game has no shopping lists.
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="_noLists_92c834"
    >
      This game has no shopping lists.
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ShoppingListGrouping > when there are shopping lists > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_92c834"
      >
        <div
          class="_shoppingList_92c834"
        >
          <div
            class="_root_62bc73"
            style="--scheme-color: #ffbf00; --border-color: #cc9800; --text-color-primary: #4c3900; --text-color-secondary: #4c3900; --hover-color: #e5ab00; --scheme-color-lighter: #ffcb32; --scheme-color-lightest: #fff2cc;"
          >
            <div
              aria-controls="list3Details"
              aria-expanded="false"
              class="_trigger_62bc73"
              role="button"
            >
              <h3
                class="_title_62bc73"
              >
                All Items
              </h3>
            </div>
            <div
              aria-hidden="true"
              class="rah-static rah-static--height-zero "
              id="list3Details"
              style="height: 0px; overflow: hidden;"
            >
              <div
                style="display: none;"
              >
                <div
                  class="_details_62bc73"
                >
                  <div
                    class="_root_fccc8d"
                    style="--main-color: #ffcb32; --title-text-color: #4c3900; --border-color: #cc9800; --body-background-color: #fff2cc; --body-text-color: #7f5700; --hover-color: #ffc519;"
                  >
                    <div
                      aria-controls="shoppingListItem6Details"
                      aria-expanded="false"
                      class="_trigger_fccc8d"
                      role="button"
                    >
                      <span
                        class="_descriptionContainer_fccc8d"
                      >
                        <h3
                          class="_description_fccc8d"
                        >
                          Dwarven Cog
                        </h3>
                      </span>
                      <span
                        class="_quantity_fccc8d"
                      >
                        11
                      </span>
                    </div>
                    <div
                      aria-hidden="true"
                      class="rah-static rah-static--height-zero "
                      id="shoppingListItem6Details"
                      style="height: 0px; overflow: hidden;"
                    >
                      <div
                        style="display: none;"
                      >
                        <div
                          class="_details_fccc8d"
                        >
                          <h4
                            class="_label_fccc8d"
                          >
                            Unit Weight:
                          </h4>
                          <p
                            class="_value_fccc8d"
                          >
                            10
                          </p>
                          <h4
                            class="_label_fccc8d"
                          >
                            Notes:
                          </h4>
                          <p
                            class="_value_fccc8d"
                          >
                            For Arniel Gane -- For trophy room
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="_root_fccc8d"
                    style="--main-color: #ffcb32; --title-text-color: #4c3900; --border-color: #cc9800; --body-background-color: #fff2cc; --body-text-color: #7f5700; --hover-color: #ffc519;"
                  >
                    <div
                      aria-controls="shoppingListItem9Details"
                      aria-expanded="false"
                      class="_trigger_fccc8d"
                      role="button"
                    >
                      <span
                        class="_descriptionContainer_fccc8d"
                      >
                        <h3
                          class="_description_fccc8d"
                        >
                          This item has a really really really really really long description for testing purposes
                        </h3>
                      </span>
                      <span
                        class="_quantity_fccc8d"
                      >
                        200000000000
                      </span>
                    </div>
                    <div
                      aria-hidden="true"
                      class="rah-static rah-static--height-zero "
                      id="shoppingListItem9Details"
                      style="height: 0px; overflow: hidden;"
                    >
                      <div
                        style="display: none;"
                      >
                        <div
                          class="_details_fccc8d"
                        >
                          <h4
                            class="_label_fccc8d"
                          >
                            Unit Weight:
                          </h4>
                          <p
                            class="_value_fccc8d"
                          >
                            400000000000
                          </p>
                          <h4
                            class="_label_fccc8d"
                          >
                            Notes:
                          </h4>
                          <p
                            class="_value_fccc8d"
                          >
                            -
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="_shoppingList_92c834"
        >
          <div
            class="_root_62bc73"
            style="--scheme-color: #e83f6f; --border-color: #b93258; --text-color-primary: #fff; --text-color-secondary: #fff; --hover-color: #d03863; --scheme-color-lighter: #ec658b; --scheme-color-lightest: #fad8e2;"
          >
            <div
              aria-controls="list4Details"
              aria-expanded="false"
              class="_trigger_62bc73"
              role="button"
            >
              <h3
                class="_title_62bc73"
              >
                Honeyside
              </h3>
            </div>
            <div
              aria-hidden="true"
              class="rah-static rah-static--height-zero "
              id="list4Details"
              style="height: 0px; overflow: hidden;"
            >
              <div
                style="display: none;"
              >
                <div
                  class="_details_62bc73"
                >
                  <div
                    class="_root_fccc8d"
                    style="--main-color: #ec658b; --title-text-color: #fff; --border-color: #b93258; --body-background-color: #fad8e2; --body-text-color: #451221; --hover-color: #ea527d;"
                  >
                    <div
                      aria-controls="shoppingListItem8Details"
                      aria-expanded="false"
                      class="_trigger_fccc8d"
                      role="button"
                    >
                      <span
                        class="_descriptionContainer_fccc8d"
                      >
                        <h3
                          class="_description_fccc8d"
                        >
                          This item has a really really really really really long description for testing purposes
                        </h3>
                      </span>
                      <span
                        class="_quantity_fccc8d"
                      >
                        200000000000
                      </span>
                    </div>
                    <div
                      aria-hidden="true"
                      class="rah-static rah-static--height-zero "
                      id="shoppingListItem8Details"
                      style="height: 0px; overflow: hidden;"
                    >
                      <div
                        style="display: none;"
                      >
                        <div
                          class="_details_fccc8d"
                        >
                          <h4
                            class="_label_fccc8d"
                          >
                            Unit Weight:
                          </h4>
                          <p
                            class="_value_fccc8d"
                          >
                            400000000000
                          </p>
                          <h4
                            class="_label_fccc8d"
                          >
                            Notes:
                          </h4>
                          <p
                            class="_value_fccc8d"
                          >
                            -
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="_root_fccc8d"
                    style="--main-color: #ec658b; --title-text-color: #fff; --border-color: #b93258; --body-background-color: #fad8e2; --body-text-color: #451221; --hover-color: #ea527d;"
                  >
                    <div
                      aria-controls="shoppingListItem5Details"
                      aria-expanded="false"
                      class="_trigger_fccc8d"
                      role="button"
                    >
                      <span
                        class="_descriptionContainer_fccc8d"
                      >
                        <h3
                          class="_description_fccc8d"
                        >
                          Dwarven Cog
                        </h3>
                      </span>
                      <span
                        class="_quantity_fccc8d"
                      >
                        1
                      </span>
                    </div>
                    <div
                      aria-hidden="true"
                      class="rah-static rah-static--height-zero "
                      id="shoppingListItem5Details"
                      style="height: 0px; overflow: hidden;"
                    >
                      <div
                        style="display: none;"
                      >
                        <div
                          class="_details_fccc8d"
                        >
                          <h4
                            class="_label_fccc8d"
                          >
                            Unit Weight:
                          </h4>
                          <p
                            class="_value_fccc8d"
                          >
                            10
                          </p>
                          <h4
                            class="_label_fccc8d"
                          >
                            Notes:
                          </h4>
                          <p
                            class="_value_fccc8d"
                          >
                            For trophy room
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="_shoppingList_92c834"
        >
          <div
            class="_root_62bc73"
            style="--scheme-color: #2274a5; --border-color: #1b5c84; --text-color-primary: #fff; --text-color-secondary: #fff; --hover-color: #1e6894; --scheme-color-lighter: #4e8fb7; --scheme-color-lightest: #d2e3ed;"
          >
            <div
              aria-controls="list5Details"
              aria-expanded="false"
              class="_trigger_62bc73"
              role="button"
            >
              <h3
                class="_title_62bc73"
              >
                Breezehome
              </h3>
            </div>
            <div
              aria-hidden="true"
              class="rah-static rah-static--height-zero "
              id="list5Details"
              style="height: 0px; overflow: hidden;"
            >
              <div
                style="display: none;"
              >
                <div
                  class="_details_62bc73"
                >
                  <div
                    class="_root_fccc8d"
                    style="--main-color: #4e8fb7; --title-text-color: #fff; --border-color: #1b5c84; --body-background-color: #d2e3ed; --body-text-color: #0d2e42; --hover-color: #3881ae;"
                  >
                    <div
                      aria-controls="shoppingListItem7Details"
                      aria-expanded="false"
                      class="_trigger_fccc8d"
                      role="button"
                    >
                      <span
                        class="_descriptionContainer_fccc8d"
                      >
                        <h3
                          class="_description_fccc8d"
                        >
                          Dwarven Cog
                        </h3>
                      </span>
                      <span
                        class="_quantity_fccc8d"
                      >
                        10
                      </span>
                    </div>
                    <div
                      aria-hidden="true"
                      class="rah-static rah-static--height-zero "
                      id="shoppingListItem7Details"
                      style="height: 0px; overflow: hidden;"
                    >
                      <div
                        style="display: none;"
                      >
                        <div
                          class="_details_fccc8d"
                        >
                          <h4
                            class="_label_fccc8d"
                          >
                            Unit Weight:
                          </h4>
                          <p
                            class="_value_fccc8d"
                          >
                            10
                          </p>
                          <h4
                            class="_label_fccc8d"
                          >
                            Notes:
                          </h4>
                          <p
                            class="_value_fccc8d"
                          >
                            For Arniel Gane
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_92c834"
    >
      <div
        class="_shoppingList_92c834"
      >
        <div
          class="_root_62bc73"
          style="--scheme-color: #ffbf00; --border-color: #cc9800; --text-color-primary: #4c3900; --text-color-secondary: #4c3900; --hover-color: #e5ab00; --scheme-color-lighter: #ffcb32; --scheme-color-lightest: #fff2cc;"
        >
          <div
            aria-controls="list3Details"
            aria-expanded="false"
            class="_trigger_62bc73"
            role="button"
          >
            <h3
              class="_title_62bc73"
            >
              All Items
            </h3>
          </div>
          <div
            aria-hidden="true"
            class="rah-static rah-static--height-zero "
            id="list3Details"
            style="height: 0px; overflow: hidden;"
          >
            <div
              style="display: none;"
            >
              <div
                class="_details_62bc73"
              >
                <div
                  class="_root_fccc8d"
                  style="--main-color: #ffcb32; --title-text-color: #4c3900; --border-color: #cc9800; --body-background-color: #fff2cc; --body-text-color: #7f5700; --hover-color: #ffc519;"
                >
                  <div
                    aria-controls="shoppingListItem6Details"
                    aria-expanded="false"
                    class="_trigger_fccc8d"
                    role="button"
                  >
                    <span
                      class="_descriptionContainer_fccc8d"
                    >
                      <h3
                        class="_description_fccc8d"
                      >
                        Dwarven Cog
                      </h3>
                    </span>
                    <span
                      class="_quantity_fccc8d"
                    >
                      11
                    </span>
                  </div>
                  <div
                    aria-hidden="true"
                    class="rah-static rah-static--height-zero "
                    id="shoppingListItem6Details"
                    style="height: 0px; overflow: hidden;"
                  >
                    <div
                      style="display: none;"
                    >
                      <div
                        class="_details_fccc8d"
                      >
                        <h4
                          class="_label_fccc8d"
                        >
                          Unit Weight:
                        </h4>
                        <p
                          class="_value_fccc8d"
                        >
                          10
                        </p>
                        <h4
                          class="_label_fccc8d"
                        >
                          Notes:
                        </h4>
                        <p
                          class="_value_fccc8d"
                        >
                          For Arniel Gane -- For trophy room
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="_root_fccc8d"
                  style="--main-color: #ffcb32; --title-text-color: #4c3900; --border-color: #cc9800; --body-background-color: #fff2cc; --body-text-color: #7f5700; --hover-color: #ffc519;"
                >
                  <div
                    aria-controls="shoppingListItem9Details"
                    aria-expanded="false"
                    class="_trigger_fccc8d"
                    role="button"
                  >
                    <span
                      class="_descriptionContainer_fccc8d"
                    >
                      <h3
                        class="_description_fccc8d"
                      >
                        This item has a really really really really really long description for testing purposes
                      </h3>
                    </span>
                    <span
                      class="_quantity_fccc8d"
                    >
                      200000000000
                    </span>
                  </div>
                  <div
                    aria-hidden="true"
                    class="rah-static rah-static--height-zero "
                    id="shoppingListItem9Details"
                    style="height: 0px; overflow: hidden;"
                  >
                    <div
                      style="display: none;"
                    >
                      <div
                        class="_details_fccc8d"
                      >
                        <h4
                          class="_label_fccc8d"
                        >
                          Unit Weight:
                        </h4>
                        <p
                          class="_value_fccc8d"
                        >
                          400000000000
                        </p>
                        <h4
                          class="_label_fccc8d"
                        >
                          Notes:
                        </h4>
                        <p
                          class="_value_fccc8d"
                        >
                          -
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="_shoppingList_92c834"
      >
        <div
          class="_root_62bc73"
          style="--scheme-color: #e83f6f; --border-color: #b93258; --text-color-primary: #fff; --text-color-secondary: #fff; --hover-color: #d03863; --scheme-color-lighter: #ec658b; --scheme-color-lightest: #fad8e2;"
        >
          <div
            aria-controls="list4Details"
            aria-expanded="false"
            class="_trigger_62bc73"
            role="button"
          >
            <h3
              class="_title_62bc73"
            >
              Honeyside
            </h3>
          </div>
          <div
            aria-hidden="true"
            class="rah-static rah-static--height-zero "
            id="list4Details"
            style="height: 0px; overflow: hidden;"
          >
            <div
              style="display: none;"
            >
              <div
                class="_details_62bc73"
              >
                <div
                  class="_root_fccc8d"
                  style="--main-color: #ec658b; --title-text-color: #fff; --border-color: #b93258; --body-background-color: #fad8e2; --body-text-color: #451221; --hover-color: #ea527d;"
                >
                  <div
                    aria-controls="shoppingListItem8Details"
                    aria-expanded="false"
                    class="_trigger_fccc8d"
                    role="button"
                  >
                    <span
                      class="_descriptionContainer_fccc8d"
                    >
                      <h3
                        class="_description_fccc8d"
                      >
                        This item has a really really really really really long description for testing purposes
                      </h3>
                    </span>
                    <span
                      class="_quantity_fccc8d"
                    >
                      200000000000
                    </span>
                  </div>
                  <div
                    aria-hidden="true"
                    class="rah-static rah-static--height-zero "
                    id="shoppingListItem8Details"
                    style="height: 0px; overflow: hidden;"
                  >
                    <div
                      style="display: none;"
                    >
                      <div
                        class="_details_fccc8d"
                      >
                        <h4
                          class="_label_fccc8d"
                        >
                          Unit Weight:
                        </h4>
                        <p
                          class="_value_fccc8d"
                        >
                          400000000000
                        </p>
                        <h4
                          class="_label_fccc8d"
                        >
                          Notes:
                        </h4>
                        <p
                          class="_value_fccc8d"
                        >
                          -
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="_root_fccc8d"
                  style="--main-color: #ec658b; --title-text-color: #fff; --border-color: #b93258; --body-background-color: #fad8e2; --body-text-color: #451221; --hover-color: #ea527d;"
                >
                  <div
                    aria-controls="shoppingListItem5Details"
                    aria-expanded="false"
                    class="_trigger_fccc8d"
                    role="button"
                  >
                    <span
                      class="_descriptionContainer_fccc8d"
                    >
                      <h3
                        class="_description_fccc8d"
                      >
                        Dwarven Cog
                      </h3>
                    </span>
                    <span
                      class="_quantity_fccc8d"
                    >
                      1
                    </span>
                  </div>
                  <div
                    aria-hidden="true"
                    class="rah-static rah-static--height-zero "
                    id="shoppingListItem5Details"
                    style="height: 0px; overflow: hidden;"
                  >
                    <div
                      style="display: none;"
                    >
                      <div
                        class="_details_fccc8d"
                      >
                        <h4
                          class="_label_fccc8d"
                        >
                          Unit Weight:
                        </h4>
                        <p
                          class="_value_fccc8d"
                        >
                          10
                        </p>
                        <h4
                          class="_label_fccc8d"
                        >
                          Notes:
                        </h4>
                        <p
                          class="_value_fccc8d"
                        >
                          For trophy room
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="_shoppingList_92c834"
      >
        <div
          class="_root_62bc73"
          style="--scheme-color: #2274a5; --border-color: #1b5c84; --text-color-primary: #fff; --text-color-secondary: #fff; --hover-color: #1e6894; --scheme-color-lighter: #4e8fb7; --scheme-color-lightest: #d2e3ed;"
        >
          <div
            aria-controls="list5Details"
            aria-expanded="false"
            class="_trigger_62bc73"
            role="button"
          >
            <h3
              class="_title_62bc73"
            >
              Breezehome
            </h3>
          </div>
          <div
            aria-hidden="true"
            class="rah-static rah-static--height-zero "
            id="list5Details"
            style="height: 0px; overflow: hidden;"
          >
            <div
              style="display: none;"
            >
              <div
                class="_details_62bc73"
              >
                <div
                  class="_root_fccc8d"
                  style="--main-color: #4e8fb7; --title-text-color: #fff; --border-color: #1b5c84; --body-background-color: #d2e3ed; --body-text-color: #0d2e42; --hover-color: #3881ae;"
                >
                  <div
                    aria-controls="shoppingListItem7Details"
                    aria-expanded="false"
                    class="_trigger_fccc8d"
                    role="button"
                  >
                    <span
                      class="_descriptionContainer_fccc8d"
                    >
                      <h3
                        class="_description_fccc8d"
                      >
                        Dwarven Cog
                      </h3>
                    </span>
                    <span
                      class="_quantity_fccc8d"
                    >
                      10
                    </span>
                  </div>
                  <div
                    aria-hidden="true"
                    class="rah-static rah-static--height-zero "
                    id="shoppingListItem7Details"
                    style="height: 0px; overflow: hidden;"
                  >
                    <div
                      style="display: none;"
                    >
                      <div
                        class="_details_fccc8d"
                      >
                        <h4
                          class="_label_fccc8d"
                        >
                          Unit Weight:
                        </h4>
                        <p
                          class="_value_fccc8d"
                        >
                          10
                        </p>
                        <h4
                          class="_label_fccc8d"
                        >
                          Notes:
                        </h4>
                        <p
                          class="_value_fccc8d"
                        >
                          For Arniel Gane
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
